File: src/main/java/ar/edu/um/haberes/rest/model/CargoLiquidacion.java
/**
 * 
 */
package ar.edu.um.haberes.rest.model;

import java.io.Serializable;
import java.math.BigDecimal;
import java.time.OffsetDateTime;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.OneToOne;

import com.fasterxml.jackson.annotation.JsonFormat;

import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

/**
 * @author daniel
 *
 */
@Entity
@EqualsAndHashCode(callSuper = false)
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class CargoLiquidacion extends Auditable implements Serializable {
	/**
	 * 
	 */
	private static final long serialVersionUID = -2029847194729942201L;

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long cargoLiquidacionId;

	private Long legajoId;
	private Integer anho;
	private Integer mes;
	private Integer dependenciaId;

	@JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssZ", timezone = "UTC")
	private OffsetDateTime fechaDesde;
	@JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssZ", timezone = "UTC")
	private OffsetDateTime fechaHasta;

	private Integer categoriaId;
	private String categoriaNombre = "";
	private BigDecimal categoriaBasico = BigDecimal.ZERO;
	private BigDecimal horasJornada = BigDecimal.ZERO;
	private Integer jornada = 0;
	private Integer presentismo = 0;
	private String situacion;

	@OneToOne
	@JoinColumn(name = "legajoId", insertable = false, updatable = false)
	private Persona persona;

	@OneToOne
	@JoinColumn(name = "dependenciaId", insertable = false, updatable = false)
	private Dependencia dependencia;

	@OneToOne
	@JoinColumn(name = "categoriaId", insertable = false, updatable = false)
	private Categoria categoria;

}

File: src/main/java/um/haberes/rest/kotlin/model/CargoLiquidacion.kt
package um.haberes.rest.kotlin.model

import com.fasterxml.jackson.annotation.JsonFormat
import jakarta.persistence.*
import java.math.BigDecimal
import java.time.OffsetDateTime

@Entity
@Table
data class CargoLiquidacion(

    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    var cargoLiquidacionId: Long? = null,
    var legajoId: Long? = null,
    var anho: Int? = null,
    var mes: Int? = null,
    var dependenciaId: Int? = null,
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssZ", timezone = "UTC")
    var fechaDesde: OffsetDateTime? = null,
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssZ", timezone = "UTC")
    var fechaHasta: OffsetDateTime? = null,
    var categoriaId: Int? = null,
    var categoriaNombre: String = "",
    var categoriaBasico: BigDecimal = BigDecimal.ZERO,
    var horasJornada: BigDecimal = BigDecimal.ZERO,
    var jornada: Int = 0,
    var presentismo: Int = 0,
    var situacion: String? = null,

    @OneToOne
    @JoinColumn(name = "legajoId", insertable = false, updatable = false)
    var persona: Persona? = null,

    @OneToOne
    @JoinColumn(name = "dependenciaId", insertable = false, updatable = false)
    var dependencia: Dependencia? = null,

    @OneToOne
    @JoinColumn(name = "categoriaId", insertable = false, updatable = false)
    var categoria: Categoria? = null

) : Auditable()
